packages/@aws-cdk/aws-ec2/lib/launch-template.ts

```diff
@@ -639,6 +707,7 @@ export class LaunchTemplate extends Resource implements ILaunchTemplate, iam.IGr
         securityGroupIds: securityGroupsToken,
         tagSpecifications: tagsToken,
         userData: userDataToken,
+        metadataOptions: this.renderMetadataOptions(props),
```

Comment:
Can you add some logic here to account for the `requireImdsv2` prop and associated aspect? Maybe we don't need to use the aspect anymore and can just set the values.